nixos/lib/test-driver/test-driver.py

```diff
@@ -675,6 +676,26 @@ def screen_matches(last: bool) -> bool:
         with self.nested("waiting for {} to appear on screen".format(regex)):
             retry(screen_matches)
 
+    def wait_for_console_text(self, regex: str) -> None:
```

Comment:
can you please document this and the OCR-using function in a way that the user understands that both basically do the same but also both exist for good reasons?

Reply:
also, in the other function i could match over lines while this one matches per-line. this is something that we either want to unify or document.

Reply:
I added a small description in `writing-nixos-test.xml`, should I add a comment in the script or document it somewhere else?

Reply:
> also, in the other function i could match over lines while this one matches per-line. this is something that we either want to unify or document.

Right, it's better to keep the two equal. I'll see what I can do to make it work on a "screen".